monai/bundle/scripts.py

```diff
@@ -822,6 +822,56 @@ def verify_metadata(
     logger.info("metadata is verified with no error.")
 
 
+def _get_net_io_info(
+    parser: ConfigParser | None = None, prefix: str = "_meta_#network_data_format", p: int = 1, n: int = 1, any: int = 1
+) -> list[Any]:
+    """
+    Get the input and output information defined in the metadata.
+
+    Args:
+        parser: a parser of the given bundle.
+        prefix: a prefix for the input and output keys, which will be combined as `prefix#input` and
+            `prefix#output` to parse the input and output information in the `metadata.json` file of
+            bundle, default to `meta_#network_data_format`.
+        p: power factor to generate input data shape if dim of expected shape is "x**p", default to 1.
+        n: multiply factor to generate input data shape if dim of expected shape is "x*n", default to 1.
+        any: specified size to generate input data shape if dim of expected shape is "*", default to 1.
+
+    Return:
+        a sequence contains the input and output infomation.
+    """
+    if not isinstance(parser, ConfigParser):
+        raise AttributeError(f"'parser' should be a ConfigParser, got {type(parser)}.")
+
+    inputs_shape = []
+    try:
+        prefix_key = f"{prefix}#inputs"
+        inputs_info_dict = parser[prefix_key]
+        for input_name in inputs_info_dict:
+            key = f"{prefix_key}#{input_name}#num_channels"
+            input_channels = parser[key]
+            key = f"{prefix_key}#{input_name}#spatial_shape"
+            input_spatial_shape = parser[key]
+            spatial_shape = _get_fake_spatial_shape(input_spatial_shape, p=p, n=n, any=any)
+            inputs_shape.append([1, input_channels, *spatial_shape])
+
+        key = f"{prefix_key}#image#num_channels"
+        input_channels = parser[key]
+        key = f"{prefix_key}#image#spatial_shape"
+        input_spatial_shape = tuple(parser[key])
+        key = f"{prefix_key}#image#dtype"
+        input_dtype = get_equivalent_dtype(parser[key], torch.Tensor)
+
+        prefix_key = f"{prefix}#outputs"
+        key = f"{prefix_key}#pred#num_channels"
+        output_channels = parser[key]
+        key = f"{prefix_key}#pred#dtype"
+        output_dtype = get_equivalent_dtype(parser[key], torch.Tensor)
+    except KeyError as e:
+        raise KeyError(f"Failed to parse due to missing expected key in the config: {key}.") from e
+    return [input_channels, input_spatial_shape, input_dtype, output_channels, output_dtype, inputs_shape]
```

Comment:
Please return tuple instead of list.

Thanks.

Reply:
What's the last item "inputs_shape" mean?

Thanks.

Reply:
It's a list of shapes of inputs. For example, if you have a model with two inputs, whose shapes are (1, 3, 224, 224) and (1, 3, 256, 256), then this item should be like `[[1,3,224,224], [1,3,256,256]]`.